src/core/Akka/Actor/ActorRef.cs

```diff
@@ -186,7 +186,8 @@ public void Tell(object message, IActorRef sender)
 
         public override string ToString()
         {
-            return string.Format("[{0}]", Path);
+            if(Path.Uid == ActorCell.UndefinedUid) return $"[{Path}]";
+            return $"[{Path}#{Path.Uid}]";
```

Comment:
UIDs are supposed to be included in the `IActorRef.ToString()` output - so you can determine from the logs if an actor was recreated or not. https://github.com/akka/akka/blob/65a9e0a0f9372d2e2ecb945c8a9dae129a438f77/akka-actor/src/main/scala/akka/actor/ActorRef.scala#L159